akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -351,6 +351,22 @@ trait FlowOps[+Out, +Mat] {
   private final val _identity = (x: Any) ⇒ x
 
   /**
+   * Recover allows to send last element on failure and gracefully complete the stream
+   * Using recover does not prevent from losing messages that arrived before onError
+   * as they pass through this processing step.
+   *
+   * '''Emits when''' element is available from the upstream or upstream is failed and pf returns an element
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' upstream completes or upstream failed with exception pf can handle
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   */
+  def recover[T >: Out](pf: PartialFunction[Throwable, T]): Repr[T, Mat] = andThen(Recover(pf.asInstanceOf[PartialFunction[Any, Any]]))
```

Comment:
RecoverWith source  can be constructed with FlexiMerge junction. It will continue from alternative source after error.
Would rather have this as a separate ticket.